### PR TITLE
[WIP] ci: use Nix shell to provide build dependencies

### PR DESCRIPTION
**Pull request by jakubgs**
Because we need Rust version newer than `1.62.0` and Ubuntu 22.04 only comes with `1.61.0`:
https://packages.ubuntu.com/jammy/rustc

And locking dependencies using Nix would make it easier for developers to track them and update them:
https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/rust/1_63.nix

### CODE REVIEW

**Comment by jakubgs** [comment]
DO NOT MERGE THIS YET. I need to tag the `status-jenkins-lib` commit once I merge the PR.

**Review by jm-clius — APPROVED** [review]

**Pull request merged**
